bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/gmos/GmosRule.java

```diff
@@ -431,6 +431,35 @@ public IConfigMatcher getMatcher() {
         }
     };
 
+    private static IConfigRule GMOS_S_DTA_X_RULE = new IConfigRule() {
+
+        private static final String Y1   = "The GMOS-S Hamamatsu allowed DTA X range is -4 to +6 for Ybin=1";
+        private static final String Y2_4 = "The GMOS-S Hamamatsu allowed DTA X range is -2 to +6 for Ybin=2 or 4";
+
+        public Problem check(Config config, int step, ObservationElements elems, Object state) {
+            final Binning y = (Binning) SequenceRule.getInstrumentItem(config, InstGmosCommon.CCD_Y_BIN_PROP);
+            final DTAX dtaX = (DTAX) SequenceRule.getInstrumentItem(config, InstGmosCommon.DTAX_OFFSET_PROP);
+
+            final ISPProgramNode node = SequenceRule.getInstrumentOrSequenceNode(step, elems, config);
+            if ((y == Binning.ONE) && (dtaX.intValue() < -4)) {
+                return new Problem(ERROR, PREFIX + "Y1", Y1, node);
+            } else if ((y != Binning.ONE) && (dtaX.intValue() < -2)) {
+                return new Problem(ERROR, PREFIX + "Y2_4", Y2_4, node);
+            } else {
+                return null;
+            }
+        }
+
+        public IConfigMatcher getMatcher() {
+            return (config, step, elems) -> {
+                final InstGmosCommon gmos = (InstGmosCommon) elems.getInstrument();
+
+                return (gmos.getSite() == Site.SET_GS) &&
```

Comment:
You're right thanks for pointing that out.  I'm under the influence of Scala.  It happens to work in this case because they refer to the same constants but that wasn't the intention.  I'll fix it before I merge.